widget/playground/src/containers/StyleLayout/StyleLayout.Preset.tsx

```diff
@@ -0,0 +1,160 @@
+import type { ColorsTypes, PresetTypes } from './StyleLayout.types';
+import type { WidgetColors } from '@rango-dev/widget-embedded';
+
+import { Button, Divider, Switch, Typography } from '@rango-dev/ui';
+import React, { useState } from 'react';
+
+import { CUSTOM_THEMES } from '../../constants';
+import { useConfigStore } from '../../store/config';
+import { shallowEqual } from '../../utils/common';
+
+import {
+  Collapse,
+  Color,
+  ColorsContent,
+  PresetContent,
+  PresetTheme,
+  Row,
+} from './StyleLayout.styles';
+
+const Colors = (props: ColorsTypes) => {
+  return (
+    <ColorsContent>
+      <Color style={{ backgroundColor: props.primary }} />
+      <Color
+        style={{
+          backgroundColor: props.secondary,
+          position: 'absolute',
+        }}
+      />
+      <Color
+        style={{
+          backgroundColor: props.background,
+          position: 'relative',
+        }}
+      />
+    </ColorsContent>
+  );
+};
+
+const FIX_HEIGHT = 41;
+const NUMBER_ITEMS_2_COL = 4;
+const NUMBER_ITEMS_1_COL = 2;
+
+export function Preset(props: PresetTypes) {
+  const { tab } = props;
+  const [showMore, setShowMore] = useState(true);
+  const onChangeTheme = useConfigStore.use.onChangeTheme();
+  const { theme } = useConfigStore.use.config();
+  const onSelectTheme = useConfigStore.use.onSelectTheme();
+  const CUSTOM_THEMES_FILTER = CUSTOM_THEMES.filter(
+    (theme) =>
+      ((tab === 'dark' || tab === 'light') && Object.keys(theme[tab]).length) ||
+      (tab === 'auto' &&
+        Object.keys(theme.dark).length &&
+        Object.keys(theme.light).length)
+  );
+  const lengthCustomThemes = CUSTOM_THEMES_FILTER.length;
+  const more =
+    lengthCustomThemes -
+    (tab === 'auto' ? NUMBER_ITEMS_1_COL : NUMBER_ITEMS_2_COL);
+
+  const height = showMore
+    ? FIX_HEIGHT * 2
+    : FIX_HEIGHT *
+      (tab === 'auto' ? lengthCustomThemes : Math.ceil(lengthCustomThemes / 2));
+
+  const onSelectPreset = (customTheme: {
+    dark: WidgetColors;
+    light: WidgetColors;
+  }) => {
+    onSelectTheme(customTheme);
+    if (tab === 'auto') {
+      onChangeTheme({ name: 'singleTheme', value: false });
+    } else {
+      onChangeTheme({ name: 'singleTheme', value: true });
+    }
+  };
+
+  return (
+    <>
+      {tab === 'auto' && (
+        <>
+          <Row>
+            <Typography variant="label" size="medium" color="neutral900">
+              Show Widget in Dark Theme
+            </Typography>
+            <Switch
+              checked={theme?.mode === 'dark'}
+              onChange={(checked) =>
+                onChangeTheme({
+                  name: 'mode',
+                  value: checked ? 'dark' : 'light',
+                })
+              }
+            />
+          </Row>
+          <Divider size={12} />
+        </>
+      )}
+      <Collapse style={{ height }}>
+        <PresetContent
```

Comment:
Please reuse `isAutoTab` variable in next lines.

Reply:
fixed